prometheus/src/main/scala/filodb/prometheus/ast/Vectors.scala

```diff
@@ -265,9 +267,23 @@ sealed trait Vector extends Expression {
         case NotRegexMatch   => require(labelValue.length <= Parser.REGEX_MAX_LEN,
                                          s"Regular expression filters should be <= ${Parser.REGEX_MAX_LEN} characters")
                                 ColumnFilter(labelMatch.label, query.Filter.NotEqualsRegex(labelValue))
-        case RegexMatch      => require(labelValue.length <= Parser.REGEX_MAX_LEN,
-                                         s"Regular expression filters should be <= ${Parser.REGEX_MAX_LEN} characters")
-                                ColumnFilter(labelMatch.label, query.Filter.EqualsRegex(labelValue))
+        case RegexMatch      =>
+          // Relax the length limit only for matchers that contain at most the "|" special character.
+          val shouldRelax = queryConfig.getBoolean("should-relax-pipe-only-equals-regex-limit") &&
```

Comment:
wondering if we can consider merging the two flags to a single flag for simplicity ? We can get the desired functionality just by `relaxed-pipe-only-equals-regex-limit` flag

```
val pipeOnlyCharLimit : Option[int] = 
if queryConfig.hasPath("relaxed-pipe-only-equals-regex-limit") {
 queryConfig.getInt("relaxed-pipe-only-equals-regex-limit");
}
else {
  None
}

val shouldRelax = QueryUtils.isPipeOnlyRegex && pipeOnlyCharLimit != None);
```

Reply:
That is way better (especially since there are already so many configs) -- I will combine these.